### ISSUE TITLE
VSCode only opens "localhost:9003" (not the correct REMOTE url) when clicking "Open in Browser" or "Preview in Editor"

### ISSUE THREAD

**Issue by RagingBullDurham** [xdebug/vscode-php-debug#974]
Xdebugging PHP code on remote VPS Server (AlmaLinux 9, PHP 8.2, Xdebug 3.2.1) from Desktop (Debian, VSCode 1.93 w/ Open Remote SSH & PHP Debug v1.22.0 extensions)

If I debug (F5) with the "Listen for Xdebug" configuration option, I get a VSCode pop up that gives me 2 buttons: "Open in Browser" and "Preview in Editor".  [Open in Browser/Preview in Editor Image](https://ibb.co/0tvQtSy)

**Pushing either option button opens up "localhost:9003" and "hangs" the debug process. I want either/both buttons to open "mysite.xxx/pageImDebugging.php)**

["Open in Browser" shows localhost:9003 Image](https://ibb.co/HBXV3Dv)

["Preview in Editor" shows localhost:9003 Image](https://ibb.co/TH5Z1hZ)

If I then manually change the url in either case to "mysite.xxx/pageImDebugging.php", the debugging process continues/works as expected.

**What VSCode/Extension/... settings can I change to get VSCode to open the correct URL?**

SERVER XDEBUG Settings: extra.ini
```
xdebug.mode = debug
xdebug.start_with_request = yes
xdebug.log_level = 10
xdebug.log = /home/USERNAME/dev/xdebug.log
```
**Open Remote - SSH Extension VSCode config file:**
```
Host MYSITE
    HostName XXX.XXX.XXX.XXX
    User USERNAME
    Port 22
    LocalForward 9003 localhost:9003
```
**XDebug extension launch.json**
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "port": 9003,
            "pathMappings": {
                "/home/USERNAME/dev": "${workspaceFolder}"
            },
        },
.. there are also default "Launch currently open script" and "Launch Built-in web server" configurations (not used for this issue)
}
```

**Comment by zobo** [comment]
Hi. Sorry for the late reply.
What you are seeing is VSCode offering you automatic port forwarding. What is actually running on port 9003 is this debug adapter. It needs to open a TCP port so that PHP/Xdebug can connect to it. So if you try to connect to it with a browser, it will not work - as there is no HTTP protocol on it.

Effectively the auto port forward for 9003 and your `LocalForward 9003 localhost:9003` do the same thing. Neither is necessary, but a change in xdebug.ini would be needed `xdebug.client_host=localhost`

Also, the `pathMappings` are probably unnecessary since VSCode is probably running on the same machine as you web server + php + xdebug.

Unless you are doing things with docker.

Basically, just ignore 9003 auto forwarded port.
